frontend/src/sagas/schedule.js

```diff
@@ -0,0 +1,441 @@
+    FULL_SCHEDULE_URL,
+    GROUPS_URL,
+    GROUP_SCHEDULE_URL,
+    PUBLIC_SEMESTERS_URL,
+    PUBLIC_TEACHER_URL,
+    ROOMS_AVAILABILITY,
+    SCHEDULE_CHECK_AVAILABILITY_URL,
+    SCHEDULE_ITEMS_URL,
+    SCHEDULE_ITEM_ROOM_CHANGE,
+    SCHEDULE_SEMESTER_ITEMS_URL,
+    SEMESTERS_URL,
+    SEND_PDF_TO_EMAIL,
+    TEACHER_SCHEDULE_URL,
+    TEACHER_URL,
+} from '../constants/axios';
+import {
+    checkAvailabilitySchedule,
+    deleteItemFromSchedule,
+    setCurrentSemester,
+    setDefaultSemester,
+    setFullSchedule,
+    setGroupSchedule,
+    setLoading,
+    setOpenSnackbar,
+    setScheduleItems,
+    setScheduleLoading,
+    setSemesterList,
+    setSemesterLoading,
+    setTeacherRangeSchedule,
+    setTeacherSchedule,
+    showAllGroups,
+    showAllTeachers,
+} from '../actions';
+import { showBusyRooms } from '../services/busyRooms';
+import { snackbarTypes } from '../constants/snackbarTypes';
+import {
+    COMMON_SCHEDULE_TITLE,
+    NO_CURRENT_SEMESTER_ERROR,
+} from '../constants/translationLabels/common';
+import {
+    BACK_END_SUCCESS_OPERATION,
+    CHOSEN_SEMESTER_HAS_NOT_GROUPS,
+    CLEARED_LABEL,
+    SERVICE_MESSAGE_GROUP_LABEL,
+    SERVICE_MESSAGE_SENT_LABEL,
+    UPDATED_LABEL,
+} from '../constants/translationLabels/serviceMessages';
+import { sortGroup } from '../helper/sortGroup';
+import {
+    FORM_CHOSEN_SEMESTER_LABEL,
+    FORM_SCHEDULE_LABEL,
+} from '../constants/translationLabels/formElements';
+import { getAllTeachersByDepartmentId } from '../actions/teachers';
+import { setMainScheduleLoading } from '../actions/loadingIndicator';
+
+export function* getScheduleItemsBySemester({ semesterId }) {
+    const requestUrl = `${SCHEDULE_SEMESTER_ITEMS_URL}?semesterId=${semesterId}`;
+    try {
+        const response = yield call(axiosCall, requestUrl);
+        yield put(setScheduleItems(response.data));
+        yield put(setScheduleLoading(false));
+    } catch (error) {
+        const message = error.response
+            ? i18n.t(error.response.data.message, error.response.data.message)
+            : 'Error';
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setScheduleLoading(false));
+    }
+}
+
+export function* getScheduleItems() {
+    try {
+        const response = yield call(axiosCall, CURRENT_SEMESTER_URL);
+        yield put(setCurrentSemester(response.data));
+        yield call(showBusyRooms, response.data.id);
+        yield call(getScheduleItemsBySemester, { semesterId: response.data.id });
+    } catch (error) {
+        const message = i18n.t(NO_CURRENT_SEMESTER_ERROR);
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setLoading(false));
+    }
+}
+
+export function* addItemsToSchedule({ item }) {
+    try {
+        yield call(axiosCall, SCHEDULE_ITEMS_URL, 'POST', item);
```

Comment:
@Akr1s 
maybe to create constants for POST...

Reply:
created